dbms/src/IO/parseDateTimeBestEffort.cpp

```diff
@@ -360,7 +361,12 @@ ReturnType parseDateTimeBestEffortImpl(time_t & res, ReadBuffer & in, const Date
                 ++in.position();
 
                 /// Just ignore fractional part of second.
-                readDigits(digits, sizeof(digits), in);
+                num_digits = readDigits(digits, sizeof(digits), in);
+                if (fractional)
+                {
+                    using FractionalType = typename std::decay<decltype(*fractional)>::type;
+                    readDecimalNumber<std::numeric_limits<FractionalType>::digits10>(*fractional, digits);
```

Comment:
It is not correct to use `std::numeric_limits<FractionalType>::digits10` here. Because in `something.11` fractional will be `1100..some amount of zeroes..00` instead of `11`.